pgcontents/crypto.py

```diff
@@ -221,8 +221,26 @@ def single_password_crypto_factory(password):
     The factory here returns a ``FernetEncryption`` that uses a key derived
     from ``password`` and salted with the supplied user_id.
     """
+    @MemoizedCryptoFactory
     def factory(user_id):
         return FernetEncryption(
             Fernet(derive_single_fernet_key(password, user_id))
         )
     return factory
+
+
+class MemoizedCryptoFactory(object):
```

Comment:
I prefer the memoization-only decorator because it's easier to test and easier to apply to other crypto factories (perhaps using different encryption). I can name it generically as you suggest.